pkg/proxy/ipvs/proxier.go

```diff
@@ -117,7 +117,6 @@ var iptablesChains = []struct {
 	{utiliptables.TableNAT, KubeNodePortChain},
 	{utiliptables.TableNAT, KubeLoadBalancerChain},
 	{utiliptables.TableNAT, KubeMarkMasqChain},
-	{utiliptables.TableNAT, KubeMarkDropChain},
```

Comment:
If I recall correctly, this is only ensuring the drop chain doesn't exist but I don't think we're writing any rules here that would conflict with kubelet. Can you point me to where we're doing this?

Reply:
lol, I've introduced this bug in kube-proxy iptables :) https://github.com/kubernetes/kubernetes/pull/85527

https://github.com/kubernetes/kubernetes/commit/9e6a6873527f2654be87fd153a8bcac969ac2d85

Reply:
Ah I see -- so as part of the save/restore it is also clearing out rules that kubelet created. 

The issue that @uablrek mentioned in https://github.com/kubernetes/kubernetes/pull/82214 still seems valid though? Can we ensure the chain exists but not try any rules to it?

Reply:
How is the dual-stack situation handled for proxy-mode=iptables? I use a work-around by creating the chain in a script before K8s is started;
```
/etc/init.d/30kube-prep.rc:   ip6tables -t nat -N KUBE-MARK-DROP
```

Reply:
Since I start `kubelet` and `kube-proxy` from scripts I also get the race. `kube-proxy` crashes and re-starts a couple of times untilt the chain is created by `kubelet`. Not a big deal, but not the most elegant solotion either.

Reply:
I think that the right fix is here https://github.com/kubernetes/kubernetes/pull/94474

Reply:
I still think it's appropriate that `kube-proxy` ensures the chain before is tries to use it but if adding it to the list also enables an undesired side-effect (sync) it must of course be corrected.

And IMHO ipvs and iptables mode should do this the same way, so the best is probably to revert the patch and wait for the fix above. Until then the chains should really be created beforehand to ensure dual-stack and, in my case, prevent the race.

Reply:
xref https://github.com/kubernetes/kubernetes/issues/82125#issuecomment-624790496
:smile:

Reply:
Thanks for reviewing 😄 

As what we did in #81517 , we chosed to belived nobody change or delete iptables chains created by kubernetes, except  system firewall be reloaded.

But we do have race issues like kube-proxy up before kubelet, maybe  ensures the chain before is tries to use it is a better way, but not try to change any rule.

After all, this is because `KUBE-MARK-DROP` chain created by kubelet and used by kube-proxy, as @aojea metioned, kube-proxy is the only consumer, maybe moving KUBE-MARK-DROP to the kube-proxies is another solution.

Overall, I think we have two potential solutions to solve this problem:
1. Ensure the chain exists but not try any rules to it (Maybe the same with `KUBE-MARK-MASQ`)
2.  Moving KUBE-MARK-DROP to the kube-proxies.

I think the first one is a better way to solve current failing test, the second way would need further discussion.

Reply:
> The issue that @uablrek mentioned in #82214 still seems valid though? Can we ensure the chain exists but not try any rules to it?

As Antonio said, the problem there is mostly that kube-proxy supports dual-stack and kubelet does not currently, and that should be fixed by #94474. Given #94474, the fix here is probably safe, although yes, I agree it would be better if ipvs could ensure that the chain exists but not delete it.

And then longer-term, #82125

Reply:
> I agree it would be better if ipvs could ensure that the chain exists but not delete it.

Can we do this now for both ipvs and iptables? Seems like the fix is to run `EnsureChain` for `KUBE-MARK-DROP` but not add anything to the nat chain buffer.

Reply:
https://github.com/kubernetes/kubernetes/blob/3fc1bc76a95d8a6a6cd33394526f00c00a98ce07/pkg/proxy/ipvs/proxier.go#L1866-L1882